loadtest/store/memstore/store.go

```diff
@@ -470,7 +480,7 @@ func (s *MemStore) SetUsers(users []*model.User) error {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	s.users = make(map[string]*model.User)
+	//s.users = make(map[string]*model.User)
```

Comment:
That commented line shows the struggle I am facing right now. I should probably add a `TODO` comment explaining what's the plan.

We currently have slightly different semantics for different store methods that should behave the same. Initially, `Set` meant replacing everything in the store with the given data. It then became more of a "update existing data and add the new one". 

The original reason to have the replace behaviour was to avoid pollution and try to keep the user's state in sync with the server. It then became clear that given that many API calls return partial data we cannot always replace but need a way to simply add/update.

I think the goal is to improve a bit the store API and make sure we have clear and consistent semantics across all store methods.

Reply:
We could have separate methods like SetUsers/PatchUsers ?

Reply:
Yes, that's a possibility. We've also have the option to pass an additional parameter to determine the behaviour of a `Set` operation if we don't want the store API to explode in the number of methods.

Ideally I'd like to get the naming right without causing too much confusion and that's why I haven't yet made any concrete proposal.